Error in GUI

**Describe the bug**  
Clicking on the Info Button of the Host on the Overview Page. After this, it opens a Window with the Error Message: Error in GUI

**Expected behavior**  
See the Information of the Host.

**Screenshots & Logfiles**  
This is the Log of the Console.
instrument.ts:132 Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7BisDocker%2C%20isOfficial%2C%20officialVersion%7D for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Ka (react-dom.production.min.js:152:60)
    at d (react-dom.production.min.js:156:186)
    at h (react-dom.production.min.js:158:331)
    at e (react-dom.production.min.js:162:334)
    at wo (react-dom.production.min.js:192:308)
    at Ei (react-dom.production.min.js:294:9)
    at ks (react-dom.production.min.js:280:389)
    at ys (react-dom.production.min.js:280:320)
    at vs (react-dom.production.min.js:280:180)
    at us (react-dom.production.min.js:271:88)
    at ss (react-dom.production.min.js:273:300)
    at Hl (react-dom.production.min.js:127:105)
    at react-dom.production.min.js:267:273
(anonymous) @ instrument.ts:132

**Versions:**  
 - Adapter version: Admin<6.7.0>
 - JS-Controller version: <5.0.11> 
 - Node version: <18.17.0>
 - Operating system: <docker>

**Additional context**  
This Problem exist after updating the Admin Adapter on Version 6.7.0

--- a/README.md
+++ b/README.md
@@ -84,6 +84,11 @@ The icons may not be reused in other projects without the proper flaticon licens
 -->
 
 ## Changelog
+### **WORK IN PROGRESS**
+* (foxriver76) fixed issue with docker host cards on intro tab
+* (foxriver76) fixed issue if adapter readme is opened in new tab
+
+
 ### 6.7.0 (2023-08-02)
 * (foxriver76) display more specific information about docker installation on host card
 * (foxriver76) allow to download folders in files tab

--- a/src/.eslintrc.js
+++ b/src/.eslintrc.js
@@ -72,5 +72,6 @@ module.exports = {
         radix: 'off',
         indent: ['error', 4, { SwitchCase: 1 }],
         'no-alert': 'off',
+        'class-methods-use-this': 'off',
     },
 };

--- a/src/package.json
+++ b/src/package.json
@@ -5,6 +5,7 @@
     "dependencies": {
         "@babel/plugin-proposal-class-properties": "^7.18.6",
         "@babel/plugin-proposal-decorators": "^7.22.7",
+        "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
         "@craco/craco": "^7.1.0",
         "@dilesoft/vite-plugin-federation-dynamic": "^1.1.9",
         "@esbuild-plugins/node-globals-polyfill": "^0.2.3",

--- a/src/src/dialogs/AdapterInfoDialog.jsx
+++ b/src/src/dialogs/AdapterInfoDialog.jsx
@@ -31,17 +31,7 @@ class AdapterInfoDialog extends Component {
     constructor(props) {
         super(props);
 
-        // const uri = `https://www.iobroker.net/${I18n.getLanguage()}/adapterref/iobroker.${props.adapter}/README.md`;
-        // this.props.link.replace('https://github.com', 'https://raw.githubusercontent.com').replace('blob/', '');
-        // const uriGithub = this.props.link.replace('https://github.com', 'https://raw.githubusercontent.com').replace('blob/', '');
-        // const rawUri = uriGithub.replace('blob/', '').substring(0, uriGithub.lastIndexOf('/') + 1);
-
         this.state = {
-        //     tab: 0,
-        //     readme: '',
-        //     uri,
-        //     rawUri,
-        //     uriGithub,
             text: null,
         };
 
@@ -56,28 +46,9 @@ class AdapterInfoDialog extends Component {
         try {
             const data = await fetch(link);
             let readme = await data.text();
-            /*
-            const lines = readme.split('\n');
-            if (lines[0].trim() === '---') {
-                let i = 1;
-                while(lines[i] !== '---') i++;
-                lines.splice(0, i + 1);
-                readme = lines.join('\n');
-                lines.unshift(`![Logo](admin/${this.props.adapter}.png)`);
-                lines.unshift('# ioBroker.' + this.props.adapter);
-                lines.unshift('');
-            }
-            const split = this.splitReadMe(readme);
-            */
+
             readme = readme.replace(/\(([-\w]+)\/adapterref\//g, '(https://www.iobroker.net/$1/adapterref/');
-            /*
-            readme = readme.replace(/\(en\/adapterref\//g, '(https://www.iobroker.net/en/adapterref/');
-            readme = readme.replace(/\(de\/adapterref\//g, '(https://www.iobroker.net/de/adapterref/');
-            readme = readme.replace(/\(zh-cn\/adapterref\//g, '(https://www.iobroker.net/zh-cn/adapterref/');
-            readme = readme.replace(/\(zh-cn\/adapterref\//g, '(https://www.iobroker.net/zh-cn/adapterref/');
-            */
             readme = readme.replace(/src="([-\w]+)\/adapterref\//g, 'src="https://www.iobroker.net/zh-cn/adapterref/');
-            // readme = readme.replace(/(<meta[^>]+>)/g, '\\$1');
 
             this.setState({ text: readme });
         } catch (error) {
@@ -85,117 +56,20 @@ class AdapterInfoDialog extends Component {
         }
     }
 
-    // static trimArr(lines) {
-    //     let j = lines.length - 1;
-    //
-    //     while (j >= 0 && !lines[j]) {
-    //         j--;
-    //     }
-    //
-    //     if (j !== lines.length - 1) {
-    //         lines.splice(j + 1);
-    //     }
-    //
-    //     return lines;
-    // }
-
-    // static splitReadMe(html, link) {
-    //     const result = {
-    //         logo: '', readme: [], changelog: [], license: [],
-    //     };
-    //     const lines = html.trim().split(/\r\n|\n/);
-    //
-    //     // second line is main title
-    //     if (lines[2].match(/^#\sio/)) {
-    //         lines.splice(2, 1);
-    //     }
-    //
-    //     if (lines[1].match(/^#\sio/)) {
-    //         lines.splice(1, 1);
-    //     }
-    //     // first line is logo
-    //     if (lines[0].match(/!\[[-_\w\d]*]\([-._\w\d/]+\.png\)/)) {
-    //         result.logo = link + lines[0].match(/\((.+)\)/)[1];
-    //         lines.splice(0, 1);
-    //     }
-    //
-    //     let part = 'readme';
-    //
-    //     for (let i = 0; i < lines.length; i++) {
-    //         if (lines[i].match(/^====/)) {
-    //             continue;
-    //         }
-    //
-    //         if (lines[i].match(/^###?\s+Changelog/)) {
-    //             part = 'changelog';
-    //             continue;
-    //         } else if (lines[i].match(/^###?\s+License/)) {
-    //             part = 'license';
-    //             continue;
-    //         } else if (lines[i].match(/^##?\s+.+/)) {
-    //             part = 'readme';
-    //         }
-    //
-    //         if (!result[part].length && !lines[i]) {
-    //             continue;
-    //         }
-    //
-    //         result[part].push(lines[i]);
-    //     }
-    //
-    //     if (result.logo) {
-    //         // that.$divLogo.html('<img src="' + result.logo + '" />').show();
-    //     } else {
-    //         // that.$divLogo.html('').hide();
-    //     }
-    //
-    //     this.trimArr(result.readme);
-    //     this.trimArr(result.changelog);
-    //     this.trimArr(result.license);
-    //
-    //     if (result.readme.length) {
-    //         result.readme = result.readme.join('\n');
-    //     } else {
-    //         result.readme = '';
-    //     }
-    //     if (result.changelog.length) {
-    //         result.changelog = result.changelog.join('\n');
-    //     } else {
-    //         delete result.changelog;
-    //     }
-    //     if (result.license.length) {
-    //         result.license[0] = `## ${result.license[0]}`;
-    //         result.license = result.license.join('\n');
-    //     } else {
-    //         delete result.license;
-    //     }
-    //
-    //     return result;
-    // }
-
-    // changeTab(event, newValue) {
-    //     this.setState({ tab: newValue });
-    // }
-
-    // openTab(path) {
-    //     const tab = window.open(path, '_blank');
-    //     tab.focus();
-    // }
+    openTab(path) {
+        const tab = window.open(path, '_blank');
+        tab.focus();
+    }
 
     static closeDialog() {
         Router.doNavigate('tab-adapters');
     }
-    //
-    // transformUri(uri) {
-    //     return (uri && uri.startsWith('http') ? '' : this.state.rawUri) + uri;
-    // }
 
     render() {
         if (!this.state.text) {
             return <Loader theme={this.props.theme} />;
         }
         const { classes } = this.props;
-        // const { tab } = this.state;
 
         return <Grid
             item
@@ -204,22 +78,6 @@ class AdapterInfoDialog extends Component {
             wrap="nowrap"
             className={classes.root}
         >
-            {/* <AppBar color="default" position="static">
-                <Tabs value={ this.state.tab } onChange={ (event, newValue) => this.changeTab(event, newValue) }>
-                    <Tab label="README" disabled={ !this.state.readme }/>
-                    <Tab label="Changelog" disabled={ !this.state.changelog }/>
-                    <Tab label="License" disabled={ !this.state.license }/>
-                </Tabs>
-            </AppBar>
-            <Box p={ 3 } className={ classes.scroll }>
-                <ReactMarkdown
-                    source={ this.state[tab === 1 ? 'changelog' : tab === 2 ? 'license' : 'readme'] }
-                    linkTarget="_blank"
-                    transformLinkUri={ (uri) => this.transformUri(uri) }
-                    transformImageUri={ (uri) => this.transformUri(uri) }
-                    escapeHtml={ false }
-                />
-            </Box> */}
             <Markdown
                 className={classes.scroll}
                 text={this.state.text}
@@ -231,10 +89,9 @@ class AdapterInfoDialog extends Component {
                 editMode={false}
                 socket={this.props.socket}
                 adapter={this.props.adapter}
-                //                https://github.com/ioBroker/ioBroker.admin/blob/master/README.md =>
+                // https://github.com/ioBroker/ioBroker.admin/blob/master/README.md =>
                 // https://raw.githubusercontent.com/ioBroker/ioBroker.admin/master/README.md
                 link={this.props.link.replace('https://github.com/', 'https://raw.githubusercontent.com/').replace('/blob/', '/')}
-                // onNavigate={(language, tab, page, chapter) => this.onNavigate(language, tab, page, chapter)}
             />
             <AppBar color="default" position="static">
                 <Toolbar>

--- a/src/src/tabs/Intro.jsx
+++ b/src/src/tabs/Intro.jsx
@@ -907,6 +907,7 @@ class Intro extends Component {
                 return this.props.socket.getCompactHosts(update);
             })
             .then(_hosts => {
+                _hosts.forEach(host => this.preprocessHostData(host));
                 hosts = _hosts;
                 return this.getInstances(update, hosts, systemConfig);
             })
@@ -957,6 +958,26 @@ class Intro extends Component {
                 {this.editLinkCard()}
             </TabContent>
         </TabContainer>;
+    }
+
+    /**
+     * Preprocess host data to harmonize information
+     *
+     * @param {Record<string, any>} hostData Host data from controller
+     * @return {Record<string, (string | number)>}
+     */
+    preprocessHostData(hostData) {
+        if (hostData.dockerInformation?.isDocker) {
+            let dockerString = hostData.dockerInformation.isOfficial ? 'official image' : 'unofficial image';
+
+            if (hostData.dockerInformation.isOfficial) {
+                dockerString +=  ` - ${hostData.dockerInformation.officialVersion}`;
+            }
+
+            hostData.Platform = `${hostData.Platform} (${dockerString})`;
+        }
+
+        delete hostData.dockerInformation;
     }
 }
 

